From 2ee1c495da28f0f3162e5a006410e09544471b54 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Mon, 21 Apr 2014 17:26:31 +0800
Subject: [PATCH 447/449] ENGR00309489 csi v4l2: correct parameter setting for
 s_fmt call

Remove s_fmt V4L2_BUF_TYPE_VIDEO_OVERLAY call.
Initialize fmt.pix with g_out_width and g_out_height.
It is the correct setting for s_fmt V4L2_BUF_TYPE_VIDEO_CAPTURE call,

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 test/mxc_v4l2_test/csi_v4l2_capture.c | 16 ++--------------
 test/mxc_v4l2_test/csi_v4l2_tvin.c    | 17 ++---------------
 2 files changed, 4 insertions(+), 29 deletions(-)

diff --git a/test/mxc_v4l2_test/csi_v4l2_capture.c b/test/mxc_v4l2_test/csi_v4l2_capture.c
index 15157c7..8b6a900 100644
--- a/test/mxc_v4l2_test/csi_v4l2_capture.c
+++ b/test/mxc_v4l2_test/csi_v4l2_capture.c
@@ -267,8 +267,8 @@ int v4l_capture_setup(void)
 	fmt.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 	fmt.fmt.pix.pixelformat = g_cap_fmt;
 	print_pixelformat("pixelformat (output by v4l)", fmt.fmt.pix.pixelformat);
-	fmt.fmt.pix.width = g_in_width;
-	fmt.fmt.pix.height = g_in_height;
+	fmt.fmt.pix.width = g_out_width;
+	fmt.fmt.pix.height = g_out_height;
 	if (ioctl(fd_v4l, VIDIOC_S_FMT, &fmt) < 0)
 	{
 		printf("set format failed\n");
@@ -286,18 +286,6 @@ int v4l_capture_setup(void)
 	}
 	g_frame_size = fmt.fmt.pix.sizeimage;
 
-	memset(&fmt, 0, sizeof(fmt));
-	fmt.type = V4L2_BUF_TYPE_VIDEO_OVERLAY;
-	fmt.fmt.win.w.left = 0;
-	fmt.fmt.win.w.top = 0;
-	fmt.fmt.win.w.width = g_out_width;
-	fmt.fmt.win.w.height = g_out_height;
-	if (ioctl(fd_v4l, VIDIOC_S_FMT, &fmt) < 0)
-	{
-		printf("set format failed\n");
-		return 0;
-	}
-
 	memset(&crop, 0, sizeof(crop));
 	crop.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 	crop.c.top = 0;
diff --git a/test/mxc_v4l2_test/csi_v4l2_tvin.c b/test/mxc_v4l2_test/csi_v4l2_tvin.c
index e5b286e..c63a1cb 100644
--- a/test/mxc_v4l2_test/csi_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/csi_v4l2_tvin.c
@@ -269,8 +269,8 @@ int v4l_capture_setup(void)
 	fmt.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 	fmt.fmt.pix.pixelformat = g_cap_fmt;
 	print_pixelformat("s_fmt input pixel format\n", fmt.fmt.pix.pixelformat);
-	fmt.fmt.pix.width = 0;
-	fmt.fmt.pix.height = 0;
+	fmt.fmt.pix.width = g_out_width;
+	fmt.fmt.pix.height = g_out_height;
 	if (ioctl(fd_v4l, VIDIOC_S_FMT, &fmt) < 0)
 	{
 		printf("set format failed\n");
@@ -288,19 +288,6 @@ int v4l_capture_setup(void)
 	}
 	g_frame_size = fmt.fmt.pix.sizeimage;
 
-
-	memset(&fmt, 0, sizeof(fmt));
-	fmt.type = V4L2_BUF_TYPE_VIDEO_OVERLAY;
-	fmt.fmt.win.w.left = 0;
-	fmt.fmt.win.w.top = 0;
-	fmt.fmt.win.w.width = g_out_width;
-	fmt.fmt.win.w.height = g_out_height;
-	if (ioctl(fd_v4l, VIDIOC_S_FMT, &fmt) < 0)
-	{
-		printf("set format failed\n");
-		return 0;
-	}
-
 	memset(&crop, 0, sizeof(crop));
 	crop.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
 	if (ioctl(fd_v4l, VIDIOC_G_CROP, &crop) < 0) {
-- 
1.8.0

